test/groupBy.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+var S = require('..');
+
+var eq = require('./internal/eq');
+
+
+test('groupBy', function() {
+
+  eq(typeof S.groupBy, 'function');
+  eq(S.groupBy.length, 2);
+  eq(S.groupBy.toString(), 'groupBy :: (a -> a -> Boolean) -> Array a -> Array (Array a)');
+
+  var productsOf3 = S.curry2(function(a, b) {
+    return a * b % 3 === 0;
+  });
+
+  eq(S.groupBy(productsOf3, []), []);
+  eq(S.groupBy(productsOf3, [1, 2, 3, 4, 5, 6, 7, 8, 9]), [[1], [2, 3], [4], [5, 6], [7], [8, 9]]);
+  eq(S.groupBy(S.equals, [1, 1, 2, 3, 3, 4]), [[1, 1], [2], [3, 3], [4]]);
+  eq(S.groupBy(S.equals, [1, 1, 2, 1, 1]), [[1, 1], [2], [1, 1]]);
+
+});
```

Comment:
Does `'nat -> nat -> bool'` generate a function which arbitrarily chooses between `true` and `false`? If so, that seems appropriate. We don't mind in this context whether the function is actually an equality function.

You could drop the number of tests from `1000` to `100` to avoid making `make test` a lot slower. :)